### DISCUSSION TITLE
Is it possible to use "keyword" in private dictionary?

### DISCUSSION THREAD

**Discussion by LoCUBUS** [fo-dicom/fo-dicom#1277] [Q&A]
Hello,

I'd like to use the "keyword" within private DICOM dictionary. Is this possible? Did somebody use it already?
The reason why I like to do this is that I need to access some private tags. like the example below:

`
<tag group="0017" element="xxCA" vr="US" vm="1" keyword="IsRejected">Image Is Rejected</tag>
`

The use of keyword would make it easier to access the private tags.

Thank you in advance for sharing your insights.

Best Regards
John

**Comment by mrbean-bremen** [comment]
I don't think that this is possible, because the private tag description is not standardized, and may not be unique. There can be the same description for tags with different private creators.

**Reply by LoCUBUS** [reply]
I fear you're right. Nevertheless I will try some ideas. I thought about giving the keywords some "ugly" pre fixes like:

keyword="SHSpri_ImageIsRejected"

Maybe it works.

**Comment by gofal** [comment]
You want to access your tags in a more comfortable way?
All the standardized Tags like `DicomTag.PatientName` are just properties that are generated into sourcecode via T4. 
You have to add these properties in your own code. So beside adding the tag into the xml dictionary

```
<dictionary creator="SHSpri">
   <tag group="0017" element="xxCA" vr="US" vm="1" keyword="IsRejected">Image Is Rejected</tag>
</dictionary
```
, you also have to add a class like

```
public static class MyTags {
   public static DicomTag ImageIsRejected = new DicomTag(0x17, 0xCA, "SHSpri");
}
```

then later in code you can easily access like `dataset.GetSingleValue<uint>(MyTags.ImageIsRejected)` just as you would access `dataset.GetSingleValue<string>(DicomTag.PatientName)`

**Comment by jingkunhub** [comment]
I'd like to also use the private tags in Deserialization from Json to Dicom or vice versa after adding tags as described in the earlier discussions of utilizing keywords and defining DicomTags, e.g., 
            const string json = "{\"IsRejected\": { \"vr\": \"CS\"Value\": [{ \"TRUE\" }] } }";
            var reconstituated = JsonConvert.DeserializeObject<DicomDataset>(json, new JsonDicomConverter(true));

By default, only keywords of DicomTag are supported. The question is how to get (De)serialization APIs support MyTags keywords.
Has anyone tried this scenario?

**Reply by jingkunhub** [reply]
Actually, I just tried this simple scenario and confirmed that DeserializeObject API works as expected and will continue to apply the same to more complex scenarios.

**Reply by jingkunhub** [reply]
Hi FO-DICOM developers,

I'm able to specify keywords in my private dictionary, use them in Dicom JSON files, and save such JSON file into DICOM file. However,
The keywords of private tags serialized in JSON file are in private tag format (setting UseKeyword to true), e.g., expecting "JhuPrivateTag" but "00111003", although the previously defined private dictionary gets loaded before serialization.

What else needs to done in order to serialize private tags in the same way as done to public tags?

Thanks in advance
